src/runtime.ffi.mjs

```diff
@@ -1,27 +1,38 @@
-import { Empty } from "./gleam.mjs";
+import { Empty, List } from "./gleam.mjs";
 import { map as result_map } from "../gleam_stdlib/gleam/result.mjs";
 
 export function morph(prev, curr, dispatch, parent) {
-  if (curr[3]) {
+  // curr is the `ElementNs` variant
+  if (curr?.[3]) {
     return prev?.nodeType === 1 &&
       prev.nodeName === curr[0].toUpperCase() &&
       prev.namespaceURI === curr[3]
       ? morphElement(prev, curr, curr[3], dispatch, parent)
       : createElement(prev, curr, curr[3], dispatch, parent);
   }
 
-  if (curr[2]) {
+  // curr is the `Element` variant
+  if (curr?.[2]) {
     return prev?.nodeType === 1 && prev.nodeName === curr[0].toUpperCase()
       ? morphElement(prev, curr, null, dispatch, parent)
       : createElement(prev, curr, null, dispatch, parent);
   }
 
+  // curr is the `Text` variant
   if (typeof curr?.[0] === "string") {
     return prev?.nodeType === 3
       ? morphText(prev, curr)
       : createText(prev, curr);
   }
 
+  // curr is the `Fragment` variant
+  if (List.isList(curr?.[0])) {
+    // I should morph/create the thingies
+    // but first I have to understand what
+    // `nodeType` is and how to use that
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType

Reply:
Thanks for the reference! It's going to be really handy